src/Middleware/RequestThrottling/src/RequestThrottlingEventSource.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+using System.Threading;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.RequestThrottling
+{
+    internal sealed class RequestThrottlingEventSource : EventSource
+    {
+        public static readonly RequestThrottlingEventSource Log = new RequestThrottlingEventSource();
+        private static QueueFrame _cachedNonTimerResult
+        {
+            get
+            {
+                if (_queueFrameCache == null)
+                {
+                    _queueFrameCache = new QueueFrame { _parent = Log };
```

Comment:
Since this is a static property this should probably be thread-safe. Usually that means using `Lazy<T>` or manually locking, but in this case I think we can just greedily initialize it like we do for the RequestThrottlingEventSource. The static initializers/ctor for the class shouldn't be run until the class is reference anyway.